### PR TITLE
Optional worldmod to disable lifting requirements (strength and lifting quality)

### PR DESCRIPTION
**Pull request by Wad67**
This is a small, optional 're-balance' mod that simply bypasses the checks for lifting quality and strength when adding or removing vehicle parts.

Adds DISABLE_LIFTING world option, mod and relevant 'overrides' to  can_install_part() and  can_remove_part( ).

### CODE REVIEW

File: src/veh_interact.cpp [hunk 1]
```diff
@@ -605,6 +605,10 @@ bool veh_interact::can_install_part() {
         str = base.lift_strength();
         use_aid = max_lift >= lvl;
         use_str = g->u.can_lift( base );
+        if( get_option<bool>( "DISABLE_LIFTING" ) ) {
```
Comment (Night-Pryanik) [inline src/veh_interact.cpp]: This check should be done earlier. I.e. if this option is active, there's no need to calculate `use_aid` and `use_str` at all.

**Comment by Night-Pryanik** [comment]
Please do tell me: when even a wimp has an ability to lift heavy vehicle parts with their bare hands, this adds something incredibly fun or interesting to the game?

File: src/veh_interact.cpp (hunk 1 above)
Comment (Wad67) [reply src/veh_interact.cpp]: Okay, I'll see what I can do.

**Comment by Wad67** [comment]
Not entirely sure what you are asking.
The purpose of this mod is to optionally remove something that people aren't interested in dealing with. 
Similar to prevent zombie revivification, simplified nutrition, it is an optional revert of a feature  #16280

**Comment by NickGrease** [comment]
Just a little nit-pic, this doesn't really revert. It removes. I'd be happy with *reverting* back to `JACK`, or calling the PR what it really is. The same way `Simplified Nutrition` doesn't remove eating.

**Comment by Wad67** [comment]
Apologies, I wasn't aware that jacking was a requirement for parts before lifting.

Unless you are referring to the jacking quality required for tires, which is left untouched.

**Comment by Wad67** [comment]
Yes, the strength check is removed also as I consider it to be a 'lifting' requirement. Note that this only effects part removals and installations. 
If I just disabled lifting, it would still be restricted(even more so by the absence of lifting equipment).

**Comment by DoctorGoat** [comment]
I'm in favor of things like this. It's good to be able to customize your game, and a customization like this won't cause problems for maintenance due to its simplicity.

**Comment by NickGrease** [comment]
Since this is an `Option` i'm not to concerned, just wanted to point out this is a removal (of Strength), and would like to see that in the PR title if only for clarity. Future searches are easier with well labeled PR's.

**Comment by Xhuis** [comment]
Agreed that this is a good thing. People come to Cata for different things - some praise its realism, and some praise the crazy stuff and gameplay. Allowing both parties to enjoy the game more fully is, in my eyes, a good thing no matter how you view it.

**Comment by Leland** [comment]
Any interest in signing as the author and maintainer of the mod?

**Comment by Wad67** [comment]
Sure, don't know where I would do that though.

**Comment by Wad67** [comment]
Okay, done.

**Comment by BorkBorkGoesTheCode** [comment]
The plus is a typo.

**Comment by Wad67** [comment]
Are you referring to the one in DracoGriffin's example, or some other '+' ?

File: data/mods/No_Lift_Requirements/modinfo.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "type": "MOD_INFO",
     "ident": "disable_lifting",
     "name": "No Lifting Reqs",
+    "authors": "Wad67",
     "description": "Strength checks and lifting qualities are no longer required for installing or removing vehicle parts.",
```
Comment (BorkBorkGoesTheCode) [inline data/mods/No_Lift_Requirements/modinfo.json]: I don't know why but copy+pasting code from GitHub's UI includes -+ signs like this and line numbers.

**Comment by DracoGriffin** [comment]
I think BorkBork was trying to point out my + was a "typo", but I was just trying make sure you saw the line. However, you added the code perfectly so there is no issue with your PR.

**Comment by BorkBorkGoesTheCode** [comment]
>I think BorkBork was trying to point out my + was a "typo", but I was just trying make sure you saw the line. However, you added the code perfectly so there is no issue with your PR.

Half correct; I also misinterpreted the plus in the changelog.

**Comment by Wad67** [comment]
It was a bit confusing given the context, I was under the impression I slipped a plus in by accident as well. A quick check proved otherwise though.

**Comment by Mecares** [comment]
I would like a non game option solution to the problem of limited lifting possibilities, like allow to disassemble a rope and pulley to make your own tripod pulley with a reasonable easy crafting recipe.

**Comment by Wad67** [comment]
Already an issue for that one, see: #23094
Granitecosmos mentioned they might have something in the works.

**Comment by kevingranade** [comment]
If it's too hard to arrange for a lifting device, just debug one in, or add
a handheld lifting device to your mod along with numerous spawn locations
for it, or add the lifting quality to heavy sticks.

I'm not interested in supporting code to disable arbitrary features
scattered throughout the code.

**Comment by Wad67** [comment]
Fair enough then.

**Pull request closed without merging**